app/lib/smaxtec_api.rb

```diff
@@ -5,21 +5,32 @@ class SmaxtecApi
   SMAXTEC_API_PASSWORD = Rails.application.secrets.smaxtec_api_password
   SMAXTEC_API_BASE_URL = 'https://api-staging.smaxtec.com/api/v1'
 
-  def get_temperature
+  def update_sensor_readings
+    Sensor.where.not(animal_id: [nil, ""]).each do |sensor|
+      # temperature sensor
+      if sensor.sensor_type.property == 'Temperature'
+        # get temperature data from smaxtec api
+        temperature = get_temperature(sensor.animal_id)
+
+        if temperature
+          # create new sensor readig for sensor
+          sensor_reading = Sensor::Reading.new(sensor_id: sensor.id, calibrated_value: temperature, uncalibrated_value: temperature)
+          sensor_reading.save!
+        end
+      end
+    end
+  end
+
+
+  def get_temperature(animal_id)
     jwt_request = get_jwt
 
     if jwt_request
       @jwt = JSON.parse(jwt_request)['token']
-      organisation_id ='5721e3f8a80a5f54c6315131';
-      animal_id = '5722099ea80a5f54c631513d' # name = Arabella
+      #animal_id = '5722099ea80a5f54c631513d' # name = Arabella
       metric = 'temp'
-      unit = 'degree_celsius'
       temp_data = send_api_request('/data/query', { :animal_id => animal_id, :metric => metric, :from_date => Time.now.to_i - 3600, :to_date => Time.now.to_i, :aggregation => 'hourly.mean' })
```

Comment:
Make sure, the new sensor reading will get some attribute that is unique both in our database and in the Smaxtec database